releases/net471/dotnet471-api-changes.md

```diff
@@ -0,0 +1,461 @@
+     }
+-    public class Tuple<T1, T2> : IComparable, IStructuralComparable, IStructuralEquatable, ITuple {
++    public class Tuple<T1, T2> : IComparable, IStructuralComparable, IStructuralEquatable, ITuple, ITupleInternal {
++        object System.Runtime.CompilerServices.ITuple.this[int index] { get; }
++        int System.Runtime.CompilerServices.ITuple.Length { get; }
+     }
+-    public class Tuple<T1, T2, T3> : IComparable, IStructuralComparable, IStructuralEquatable, ITuple {
++    public class Tuple<T1, T2, T3> : IComparable, IStructuralComparable, IStructuralEquatable, ITuple, ITupleInternal {
++        object System.Runtime.CompilerServices.ITuple.this[int index] { get; }
++        int System.Runtime.CompilerServices.ITuple.Length { get; }
+     }
+-    public class Tuple<T1, T2, T3, T4> : IComparable, IStructuralComparable, IStructuralEquatable, ITuple {
++    public class Tuple<T1, T2, T3, T4> : IComparable, IStructuralComparable, IStructuralEquatable, ITuple, ITupleInternal {
++        object System.Runtime.CompilerServices.ITuple.this[int index] { get; }
++        int System.Runtime.CompilerServices.ITuple.Length { get; }
+     }
+-    public class Tuple<T1, T2, T3, T4, T5> : IComparable, IStructuralComparable, IStructuralEquatable, ITuple {
++    public class Tuple<T1, T2, T3, T4, T5> : IComparable, IStructuralComparable, IStructuralEquatable, ITuple, ITupleInternal {
++        object System.Runtime.CompilerServices.ITuple.this[int index] { get; }
++        int System.Runtime.CompilerServices.ITuple.Length { get; }
+     }
+-    public class Tuple<T1, T2, T3, T4, T5, T6> : IComparable, IStructuralComparable, IStructuralEquatable, ITuple {
++    public class Tuple<T1, T2, T3, T4, T5, T6> : IComparable, IStructuralComparable, IStructuralEquatable, ITuple, ITupleInternal {
++        object System.Runtime.CompilerServices.ITuple.this[int index] { get; }
++        int System.Runtime.CompilerServices.ITuple.Length { get; }
+     }
+-    public class Tuple<T1, T2, T3, T4, T5, T6, T7> : IComparable, IStructuralComparable, IStructuralEquatable, ITuple {
++    public class Tuple<T1, T2, T3, T4, T5, T6, T7> : IComparable, IStructuralComparable, IStructuralEquatable, ITuple, ITupleInternal {
++        object System.Runtime.CompilerServices.ITuple.this[int index] { get; }
++        int System.Runtime.CompilerServices.ITuple.Length { get; }
+     }
+-    public class Tuple<T1, T2, T3, T4, T5, T6, T7, TRest> : IComparable, IStructuralComparable, IStructuralEquatable, ITuple {
++    public class Tuple<T1, T2, T3, T4, T5, T6, T7, TRest> : IComparable, IStructuralComparable, IStructuralEquatable, ITuple, ITupleInternal {
++        object System.Runtime.CompilerServices.ITuple.this[int index] { get; }
++        int System.Runtime.CompilerServices.ITuple.Length { get; }
+     }
+-    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential, Size=1)]
```

Comment:
I don't think that it's particularly useful to display the StructLayoutAttribute. If an attribute is to be shown, it would be better to display Serializable, which isn't shown here or in the following ValueTuple generic types; that's one of the what's new items.

Reply:
This is from APIPort export -- there isn't much a flexibility to exclude anything.